packages/fetchai/connections/p2p_libp2p/dht/dhtpeer/dhtpeer.go

```diff
@@ -119,6 +120,8 @@ type DHTPeer struct {
 	myAgentReady     func() bool
 	dhtAddresses     map[string]string
 	tcpAddresses     map[string]net.Conn
+	dhtAddressesLock sync.RWMutex
```

Comment:
probably can be replaced by one lock, but not sure

Reply:
yes, it can.
however it'll limit the degree of parallelism
otherwise, I agree that this need to be refactored to abstract locks away